### DISCUSSION TITLE
Export new AstNodeDescriptions after scoping ?

### DISCUSSION THREAD

**Discussion by antoinelilly** [eclipse-langium/langium#1757] [Q&A]
Hello all !

I would like to know if it is possible to export a new description for a node after scoping ? If not, maybe give me an idea to solve the following problem :

My grammar is based on classes and attributes with inheritance. I already implemented qualified name scoping by overriding getScope and computeExports which works well, except in one case. Consider the following : 

```
class Parent {
    attribute1: string
}

class Child extends Parent {
    attribute2: string
}

class Other {
   attribute3: Child.attribute2
   attribute4: Parent.attribute1
   attribute5: Child.attribute1
}
```

Scoping works for Child.attribute2 and Parent.attribute1 because I correctly exported the attributes descriptions under their qualified names. But for Child.attribute1, which I want be available because it is an inherited attribute from the Parent class, I don't know how to proceed. I can't export its descriptions like the others because the scoping on the Parent reference (in "extends Parent") hasn't been solved yet. That's why my idea was to create a new description for the node after scoping.

Any help would be most welcome!

**Comment by cdietrich** [comment]
is there a reason you decided against using class member scoping?

**Reply by msujew** [reply]
I agree with @cdietrich, using an index based scoping mechanism isn't suitable for this case. A more *dynamic* approach should be taken that doesn't require the full property names being available in the index. [Class member scoping](https://langium.org/docs/recipes/scoping/class-member/) is more suitable for this.

**Reply by antoinelilly** [reply]
Ok I understand, at first I chose qualified names because I also have an optional cross reference with an import alias like "alias.class.attribute" but the fact that it was optional caused a parser ambiguity.

I will use class member scoping for this specific case then thank you!
